src/node_jsvmapi_types.h

```diff
@@ -34,4 +36,48 @@ struct napi_property_descriptor {
   void* data;
 };
 
+enum napi_valuetype {
+  // ES6 types (corresponds to typeof)
+  napi_undefined,
+  napi_null,
+  napi_boolean,
+  napi_number,
+  napi_string,
+  napi_symbol,
+  napi_object,
+  napi_function,
+};
+
+enum napi_typedarray_type {
```

Comment:
I think `napi_valuetype` and `napi_get_type_of_value` are intentionally limited to fundamental JavaScript types. (Technically `napi_function` breaks that rule because a function is also an object, but I think I'm OK with that.) Adding other Object subclasses to the enum would make things confusing: `napi_get_type_of_value(myTypedArray) == napi_object` would be false, even though a typed-array is in fact an object. Perhaps that's why V8 provides lots of Is*SomeType*() methods instead.